libs/components/src/trip-form-create/TripFormCreate.tsx

```diff
@@ -0,0 +1,258 @@
+          membersToInvite={parameters.membersToInvite}
+          auth={parameters.auth}
+          setMembersToInvite={parameters.setMembersToInvite}
+        />
+      );
+    case 3:
+      return <TitleStep
+        formField={formField}
+        formValues={parameters.formValues}
+        setFieldValue={parameters.setFieldValue}
+        setFieldTouched={parameters.setFieldTouched}
+      />
+    case 4:
+      return <ImageStep
+        formField={formField}
+        formValues={parameters.formValues}
+        setFieldValue={parameters.setFieldValue}
+      />
+    default:
+      return <div>Form Not Found</div>
+  }
+};
+
+export function TripFormCreate() {
+  const auth = useSelector((state: AppState) => state.firebase.auth)
+  const profile = useSelector((state: RootState) => state.firebase.profile);
+  const activeLoggedInUser = useLoggedInUser()
+  const firebase = useFirebase();
+  const dispatch = useDispatch();
+  const router = useRouter();
+
+  const [isLoading, setIsLoading] = useState(false);
+  const [activeStep, setActiveStep] = useState(0)
+  const [membersToInvite, setMembersToInvite] = useState<MembersToInviteType>([]);
+
+  const isLastStep = activeStep === steps.length - 1;
+
+  const initialValues: TripFormType = getInitValues(auth.uid);
+  const currentValidationSchema = validationSchema[activeStep];
+
+  useFirestoreConnect([
+    {
+      collection: 'users',
+      where: [
+        'uid',
+        'in',
+        membersToInvite && membersToInvite.length > 0
+          ? membersToInvite.map((m) => m.uid)
+          : [auth.uid || ''],
+      ],
+    },
+  ])
+
+  const submitForm = (values: TripFormType) => {
+    setIsLoading(true);
+
+    const now = new Date()
+    const tripMembers: Record<string, TripMemberFormType> = {};
+    tripMembers[`${auth.uid}`] = {
+      uid: auth.uid,
+      status: TripMemberStatus.Owner,
+      invitedAt: now,
+      acceptedAt: now,
+    };
+
+    membersToInvite.forEach((member) => {
+      tripMembers[`${member.uid}`] = {
+        uid: member.uid,
+        status: TripMemberStatus.Pending,
+        invitedAt: now,
+        invitedBy: auth.uid,
+      };
+    });
+
+    firebase
+      .firestore()
+      .collection('trips')
+      .add({
+        ...values,
+        tags: [],
+        created: now,
+        tripMembers,
+      })
+      .then((docRef) => {
+        void docRef.update({
+          tripId: docRef.id,
+        });
+
+        membersToInvite.forEach((member) => {
+          void sendTripInvitationEmail({
+            tripId: docRef.id,
+            invitedBy: profile.username,
+            email: member.email,
+            greetingName: member.greetingName,
+          });
+        });
+
+        trackEvent('New Trip Submit Successful', { values: { ...values } });
+        void router.push(`/trips/${docRef.id}/generator`)
+      })
+      .catch((err) => {
+        trackEvent('New Trip Submit Unsuccessful', { values: { ...values }, error: err });
+        dispatch(
```

Comment:
i got rid of the global alerts and starting using a library instead, so this can just be `toast.error(err.message)` from `import { toast } from 'react-hot-toast'`